source/data-api.txt

```diff
@@ -0,0 +1,316 @@
+        -H 'api-key: <API Key>' \
+        -H 'Content-Type: application/json' \
+        -d '{ "message": "Hello, world!" }'
+
+The Data API is not a direct connection to your database. Instead, the
+API is a fully-managed middleware service that sits between your cluster
+and client apps. For each incoming request, the Data API authenticates
+the calling user, authorizes the request, runs the endpoint logic, and
+then returns an HTTPS response to the caller.
+
+Endpoints
+---------
+
+The Data API supports two types of endpoints:
+
+- **Data API Endpoints** are automatically generated endpoints that each
+  represent a MongoDB operation. You can use the endpoints to create,
+  read, update, delete, and aggregate documents in a MongoDB data
+  source.
+
+  To learn more, including how to set up and call endpoints, see
+  :ref:`Data API Endpoints <data-api-endpoints>`.
+
+- **Custom Endpoints** are app-specific API routes handled by functions
+  that you write. You can use custom endpoints to run your app's backend
+  logic or as webhooks that integrate with external services.
+  
+  To learn more, including how to create and call endpoints, see
+  :ref:`Custom HTTPS Endpoints <custom-endpoints>`.
+
+.. _data-api-data-formats:
+
+Data Formats
+------------
+
+MongoDB stores data in a format called BSON, which is similar to a JSON
+object in structure but supports additional data types and uses a binary
+encoding. BSON is efficient for computers but is not human readable so
+you can't work with it directly.
+
+Instead, the Data API uses two formats to represent data in requests and
+responses:
+
+- The **JSON** format uses standard types that any tool can parse and
+  understand. However, JSON cannot represent all BSON types so JSON
+  responses may lose type information for some fields. For example, BSON
+  has distinct types for 32-bit integers and 64-bit floats but a JSON
+  response represents both as a ``number``.
+
+- The **EJSON** format, short for :manual:`MongoDB Extended JSON
+  </reference/mongodb-extended-json/>`, is a superset of standard JSON
+  that uses structured fields to represent BSON data that don't have
+  corresponding JSON types. This fully represents your data but requires
+  your client to understand how to work with EJSON.
+
+.. example::
+   
+   The following document shows BSON types represented in JSON and EJSON:
+   
+   .. tabs::
+       
+       .. tab:: EJSON
+          :tabid: ejson
+          
+          .. code-block:: json
+             
+             {
+                "Name": "Mango",
+                "Year": { "$numberLong": "2022" },
+                "Weight": { "$numberDecimal": "9823.1297" },
+                "Date": { "$date": { "$numberLong": "1641954803067" } }
+             }
+       
+       .. tab:: JSON
+          :tabid: json
+          
+          .. code-block:: json
+             
+             {
+                "Name": "Mango",
+                "Year": 2022,
+                "Weight": {},
```

Comment:
why is weight an empty object, not JS number?

Reply:
The Data API "translates" EJSON into JSON, and for some types, there's no valid equivalent in JSON. In those cases the data api just returns a null/empty object (`{}`), for better or for worse.

Reply:
thx for clarification nate. then i think it's worth calling this out very clearly in an admonition b.c this is quite strange behavior that i imagine will confuse users who aren't too intimate w EJSON

Reply:
Good catch though - they updated it so Decimal128 returns the decimal as a string instead of an empty object. Updated!